### PR TITLE
Row size guardrails

### PR DESCRIPTION
**Pull request by stbof**
https://cockroachlabs.atlassian.net/browse/CRDB-8657

### CODE REVIEW

**Review by yuzefovich — COMMENTED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12247)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans, @stbof, and @vy-ton)

___
*[_includes/v21.1/sql/limit-row-size.md, line 4 at r1](https://reviewable.io/reviews/cockroachdb/docs/12247#-MoCHLZ-4K8cU0gD-Tvk:-MoCHLZ07aTRYHV1Jx1R:bm9gdzs) ([raw file](https://github.com/cockroachdb/docs/blob/49d16da549e54388ad5d9ff66bc542851b9bf971/_includes/v21.1/sql/limit-row-size.md#L4)):*
> ```Markdown
> 
> You can specify the behavior when you write a row larger than a specified size to the database (or
> individual column families, if multiple column families are in use). The [cluster settings](cluster-settings.html)
> ```

nit: maybe a link to column families docs would be useful?
___
*[_includes/v21.1/sql/limit-row-size.md, line 5 at r1](https://reviewable.io/reviews/cockroachdb/docs/12247#-MoCGwJlCtMdcNDKk2hf:-MoCGwJlCtMdcNDKk2hg:bhx1f46) ([raw file](https://github.com/cockroachdb/docs/blob/49d16da549e54388ad5d9ff66bc542851b9bf971/_includes/v21.1/sql/limit-row-size.md#L5)):*
> ```Markdown
> You can specify the behavior when you write a row larger than a specified size to the database (or
> individual column families, if multiple column families are in use). The [cluster settings](cluster-settings.html)
> `sql.mutations.max_row_size.log` and `sql.guardrails.max_row_size.err` can help you avoid failures arising
> ```

The settings ended up being named `sql.guardrails.max_row_size_log` and `sql.guardrails.max_row_size_err` after the initial PR was merged (release note in https://github.com/cockroachdb/cockroach/pull/69457).

**Review by vy-ton — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12247)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans and @stbof)

___
*[_includes/v21.2/sql/limit-row-size.md, line 3 at r1](https://reviewable.io/reviews/cockroachdb/docs/12247#-MoEr2lS6tJrLu6EJ5vz:-MoEr2lS6tJrLu6EJ5w-:b2y5hth) ([raw file](https://github.com/cockroachdb/docs/blob/49d16da549e54388ad5d9ff66bc542851b9bf971/_includes/v21.2/sql/limit-row-size.md#L3)):*
> ```Markdown
> ## Limit the size of rows
> 
> You can specify the behavior when you write a row larger than a specified size to the database (or
> ```

It's hard to follow that this setting applies to individual column families and that in CockroachDB if you don't use explicit column families `a row = implicit column family`.

The framing we want to convey here is:
* use `sql.mutations.max_row_size.log` to discover and `sql.guardrails.max_row_size.err` to enforce large rows, ex. large JSONB columns
___
*[_includes/v21.2/sql/limit-row-size.md, line 12 at r1](https://reviewable.io/reviews/cockroachdb/docs/12247#-MoErq2zCA3QsYy3xOFC:-MoErq2zCA3QsYy3xOFD:bv158wu) ([raw file](https://github.com/cockroachdb/docs/blob/49d16da549e54388ad5d9ff66bc542851b9bf971/_includes/v21.2/sql/limit-row-size.md#L12)):*
> ```Markdown
> is logged to the [`SQL_PERF`](logging.html#sql_perf) channel. `SELECT`, `DELETE`, `TRUNCATE`, and `DROP` are not affected.
> 
> When you write a row that exceeds `sql.guardrails.max_row_size.err`:
> ```

Its easier to read/follow this bullet structure, we should do the same for the `log` setting above

**Review by yuzefovich — COMMENTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12247)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans, @stbof, and @vy-ton)

___
*[_includes/v21.1/sql/limit-row-size.md, line 5 at r1](https://reviewable.io/reviews/cockroachdb/docs/12247#-MoCGwJlCtMdcNDKk2hf:-MoFbm-T9sC4qwmUxfqx:b-5wzhza) ([raw file](https://github.com/cockroachdb/docs/blob/49d16da549e54388ad5d9ff66bc542851b9bf971/_includes/v21.1/sql/limit-row-size.md#L5)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

The settings ended up being named `sql.guardrails.max_row_size_log` and `sql.guardrails.max_row_size_err` after the initial PR was merged (release note in https://github.com/cockroachdb/cockroach/pull/69457).
</blockquote></details>

Note the `_` in "size_log" and "size_err" (not "size.log" nor "size.err").

File: _includes/v21.1/sql/limit-row-size.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+When you write a row that exceeds `sql.guardrails.max_row_size_err`:
+
+- `INSERT`, `UPSERT`, and `UPDATE` statements will fail with a code `54000 (program_limit_exceeded)` error.
+
+- `CREATE TABLE AS`, `CREATE INDEX`, `ALTER TABLE`, `ALTER INDEX`, `IMPORT`, and `RESTORE` statements will log a `LargeRowInternal` event to the [`SQL_INTERNAL_PERF`](logging.html#sql_internal_perf) channel.
+
+- `SELECT`, `DELETE`, `TRUNCATE`, and `DROP` are not affected.
+
+You **cannot** update existing rows that violate the limit unless the update shrinks the size of the
+row below the limit. You **can** select, delete, alter, backed up, and restore such rows. We
```
Comment (ianjevans) [inline _includes/v21.1/sql/limit-row-size.md:19]:
```suggestion
row below the limit. You **can** select, delete, alter, back up, and restore such rows. We
```

File: _includes/v21.2/sql/limit-row-size.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+When you write a row that exceeds `sql.guardrails.max_row_size_err`:
+
+- `INSERT`, `UPSERT`, and `UPDATE` statements will fail with a code `54000 (program_limit_exceeded)` error.
+
+- `CREATE TABLE AS`, `CREATE INDEX`, `ALTER TABLE`, `ALTER INDEX`, `IMPORT`, and `RESTORE` statements will log a `LargeRowInternal` event to the [`SQL_INTERNAL_PERF`](logging.html#sql_internal_perf) channel.
+
+- `SELECT`, `DELETE`, `TRUNCATE`, and `DROP` are not affected.
+
+You **cannot** update existing rows that violate the limit unless the update shrinks the size of the
+row below the limit. You **can** select, delete, alter, backed up, and restore such rows. We
```
Comment (ianjevans) [inline _includes/v21.2/sql/limit-row-size.md:19]:
```suggestion
row below the limit. You **can** select, delete, alter, back up, and restore such rows. We
```

**Review by ianjevans — COMMENTED** [review]
Just two minor changes.

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12247#-:-MoL4t93C8tWA96p9gtv:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof and @vy-ton)

**Review by vy-ton — APPROVED** [review]

**Review by ianjevans — APPROVED** [review]
